src/pkg/Microsoft.Windows.Compatibility/Microsoft.Windows.Compatibility.pkgproj

```diff
@@ -54,54 +51,74 @@
+
     <!-- Packages we don't build in master anymore -->
     <LibraryPackage Include="System.IO.Pipes.AccessControl">
       <Version>4.5.1</Version>
     </LibraryPackage>
     <LibraryPackage Include="System.Data.DataSetExtensions">
       <Version>4.5.0</Version>
     </LibraryPackage>
+
+    <!-- Service model packages -->
+    <ExternalLibraryPackage Include="System.ServiceModel.Primitives">
+      <Version>$(ServiceModelVersion)</Version>
+    </ExternalLibraryPackage>
+    <ExternalLibraryPackage Include="System.ServiceModel.Duplex">
+      <Version>$(ServiceModelVersion)</Version>
+    </ExternalLibraryPackage>
+    <ExternalLibraryPackage Include="System.ServiceModel.Http">
+      <Version>$(ServiceModelVersion)</Version>
+    </ExternalLibraryPackage>
+    <ExternalLibraryPackage Include="System.ServiceModel.NetTcp">
+      <Version>$(ServiceModelVersion)</Version>
+    </ExternalLibraryPackage>
+    <ExternalLibraryPackage Include="System.ServiceModel.Security">
+      <Version>$(ServiceModelVersion)</Version>
+    </ExternalLibraryPackage>
   </ItemGroup>
   <ItemGroup>
     <!-- The shims included in the shims package are only needed for netcoreapp2.0 since in newer frameworks they are included inbox.
     In order to condition the shims package to an specific framework other dependencies need to specify their target framework. So
     we need to include the dependencies with the supported frameworks and then we include the compatibility shims package only for 2.0 -->
-    <Dependency Include="@(LibraryPackage);@(PrereleaseLibraryPackage)">
+    <IndexedDependency Include="@(LibraryPackage);@(PrereleaseLibraryPackage)">
+      <TargetFramework>netcoreapp2.0</TargetFramework>
+    </IndexedDependency>
+    <IndexedDependency Include="@(LibraryPackage);@(PrereleaseLibraryPackage)">
+      <TargetFramework>netcoreapp2.1</TargetFramework>
+    </IndexedDependency>
+    <IndexedDependency Include="@(LibraryPackage);@(PrereleaseLibraryPackage);@(NS21PrereleaseLibraryPackage)">
+      <TargetFramework>netcoreapp3.0</TargetFramework>
+    </IndexedDependency>
+    <IndexedDependency Include="@(LibraryPackage);@(PrereleaseLibraryPackage)">
+      <TargetFramework>netstandard2.0</TargetFramework>
+    </IndexedDependency>
+    <IndexedDependency Include="@(LibraryPackage);@(PrereleaseLibraryPackage);@(NS21PrereleaseLibraryPackage)">
+      <TargetFramework>netstandard2.1</TargetFramework>
+    </IndexedDependency>
+    <!-- Include dependency to shims package -->
+    <IndexedDependency Include="Microsoft.Windows.Compatibility.Shims">
+      <Version>2.0.1</Version>
+      <TargetFramework>netcoreapp2.0</TargetFramework>
+    </IndexedDependency>
+  </ItemGroup>
+
+  <ItemGroup>
+    <Dependency Include="@(ExternalLibraryPackage)">
       <TargetFramework>netcoreapp2.0</TargetFramework>
     </Dependency>
-    <Dependency Include="@(LibraryPackage);@(PrereleaseLibraryPackage)">
+    <Dependency Include="@(ExternalLibraryPackage)">
```

Comment:
@ericstj if I just specify this dependency as non-framework dependent, would it be correct? I mean would it support all frameworks? or should we leave as is?

Reply:
It needs to stay as is.  Remember how NuGet dependency groups work: you only get one.  So if you put this in the TFM-less dependency group it would only apply when no other dependency groups apply.

Reply:
Ok, thanks for the reminder 😄